### PR TITLE
Add Main menu

### PR DESCRIPTION
**Pull request by MadTechnocrat**

### CODE REVIEW

File: src/Core/Application.cpp [hunk 1]
```diff
@@ -11,6 +11,6 @@ void Application::Run()
 
 	while(true)
 	{
-		
+		//XD
```
Comment (RooTender) [inline src/Core/Application.cpp:14]: Please remove that.

File: src/Core/MainMenu.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+void MainMenu::InitializeGlfw()
+{
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE); //(core profile (use only modern functions to hell with old ones))
+}
+
+
+void MainMenu::Draw()
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:39]: Remove redundant comments

File: src/Core/MainMenu.cpp [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		glClear(GL_COLOR_BUFFER_BIT);
+
+		//IMGUI
+		ImGui_ImplOpenGL3_NewFrame();
+		ImGui_ImplGlfw_NewFrame();
+		ImGui::NewFrame();
+		//ImGui::SetNextWindowSize(ImVec2(600, 500));
+
+
+		//For Options:
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:89]: If you want to keep this, please write this down in the notepad

File: src/Core/MainMenu.cpp [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		glfwTerminate();
+	}
+
+	glfwMakeContextCurrent(window); //introduce new window to current context
+
+	gladLoadGL(); //load GLAD so it configures OPENGL
+	//glViewport(0, 0, 800, 800); //area of the window for opengl speicfy the viewport from 0 to 800
+
+
+	///////////////IMGUI/////////////////
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:61]: Extract that to the init function like `InitImgui`

File: src/Core/MainMenu.cpp [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE); //(core profile (use only modern functions to hell with old ones))
+}
+
+
+void MainMenu::Draw()
+{
+	//initialize GLFW
+	glfwInit();
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:42]: This should be moved to the `void MainMenu::InitializeGlfw()` _line 31_

File: src/Core/MainMenu.cpp [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		//ImGui::SetCursorPos(ImVec2(20, 20));
+
+
+		ImGui::Begin("Main Menu");
+		DrawCenteredText("Welcome to the Block Forge!");
+		DrawWindowSizedButton("PLAY");
+		DrawWindowSizedButton("OPTIONS");
+		quit = DrawWindowSizedButton("EXIT");
+
+		if(DrawWindowSizedButton("CREDITS"))
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:103]: Empty if statement. If you want to keep it, add **TODO** comment

File: src/Core/MainMenu.cpp [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+		ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+
+
+		//swap the front buffer with the back buffer
+		glfwSwapBuffers(window); 
+		//take care of all GLFW EVENTS
+		glfwPollEvents(); //process windows events: resize and so on... without this nothing will work
+	}
+
+	ImGui_ImplOpenGL3_Shutdown();
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:120]: This could be extracted to the method

File: src/Core/MainMenu.cpp [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+void MainMenu::Draw()
+{
+	//initialize GLFW
+	glfwInit();
+
+	//hints (information about version to glfw) //version 3.3
+	InitializeGlfw();
+
+	GLFWwindow* window = glfwCreateWindow(1600, 800, "Main menu", /*glfwGetPrimaryMonitor()*/  nullptr, nullptr); //width, height, name, FULLSCREAN(nullptr/?), "not important"
```
Comment (RooTender) [inline src/Core/MainMenu.cpp:47]: 1600 and 800 are magic numbers. Please add constant variables like or even better - a struct like
```
struct window {
    unsigned width = 1280;
    unsigned height = 720; 
};
```

Then use these variables here.

Comment (RooTender) [reply src/Core/MainMenu.cpp:47]: Also these comments worsen code readability

**Review by RooTender — CHANGES_REQUESTED** [review]
Please make changes

File: src/Core/Application.cpp (hunk 1 above)
Comment (MadTechnocrat) [reply src/Core/Application.cpp:14]: Deleted

File: src/Core/MainMenu.cpp (hunk 8 above)
Comment (RooTender) [reply src/Core/MainMenu.cpp]: Won't fix

**Review by RooTender — APPROVED** [review]

**Pull request merged**